src/handlers/expiry/providers/DateProvider.ts

```diff
@@ -24,14 +24,22 @@ export class DateProvider {
     this.testDate = moment(date).startOf("day").toDate();
   }
 
-//#region Private Static Functions
+  //#region Static Functions
   public static getEpoc(): Date {
-  return moment(new Date(0)).startOf("day").toDate();
+    return moment(new Date(0)).startOf("day").toDate();
   }
+
   public static isSameAsEpoc(inputDate: Date): boolean {
     return moment(inputDate).isSame(DateProvider.getEpoc());
   }
 
+  public static getInstance(dateValue: string | number | Date) {
```

Comment:
can it receive a number or Date as input? I can that getInstance is called after the date is validated with isValidDate and this function checks for 3 patterns. How would the isValidate function behaves if it receives other formats than the ones we explicitly check?
I wonder whether we should be adding unit tests for this function which will act as some documentation for other developers